jbehave-core/src/main/java/org/jbehave/core/steps/ParameterConverters.java

```diff
@@ -572,29 +587,43 @@ public ParameterConversionFailed(String message, Throwable cause) {
         public AbstractParameterConverter() {
         }
 
-        public AbstractParameterConverter(Type targetType) {
-            super(targetType);
+        public AbstractParameterConverter(Type sourceType, Type targetType) {
```

Comment:
isn't `sourceType` always equal to `String.class`  here?

Reply:
mm yes, that's why Functional parameter converter passes String.class here, I would prefer to keep this class decoupled from logic in Functional param converter

Reply:
`... extends AbstractChainableParameterConverter<String, T> ...`